ci/test/03_test_script.sh

```diff
@@ -132,6 +132,61 @@ bash -c "${PRINT_CCACHE_STATISTICS}"
 du -sh "${DEPENDS_DIR}"/*/
 du -sh "${PREVIOUS_RELEASES_DIR}"
 
+if [[ ${BARE_METAL_RISCV} == "true" ]]; then
+    cmake --build . --target bitcoin_consensus
+
+    echo -e "#include <script/script_error.h>\n int main() { return ScriptErrorString(ScriptError_t::SCRIPT_ERR_UNKNOWN_ERROR).size() > 0; }" > test.cpp
+
+    /opt/riscv-ilp32/bin/riscv32-unknown-elf-g++ -I "${BASE_ROOT_DIR}"/src -g -std=c++20 -march=rv32gc -mabi=ilp32 -c test.cpp -o test.o
+
+    echo -e ".section .text
+          .global _start
+          .type _start, @function
+
+          _start:
+              .option push
+              .option norelax
+              la gp, __global_pointer$
+              .option pop
+
+              call main
+
+              # Put Exit2 system call number into the a7 register
```

Comment:
concept ACK on making it actually runnable
i was confused for a moment here, as to what syscall numbers mean for bare-metal
maybe add "Linux" to the comment